### PR TITLE
[MAINTENANCE] Remove legacy usage stats

### PR DESCRIPTION
**Pull request by jcampbell**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

**Review by tannerbeam — COMMENTED** [review]
concept approved (did not test the branch)

**Review by tannerbeam — APPROVED** [review]

File: CODEOWNERS [hunk 1]
```diff
@@ -1,5 +1,3 @@
 # Each line is a file pattern followed by one or more owners.
 
-sidebars.js @great-expectations/fe-cop-codeowners
-
-great_expectations/core/usage_statistics/schemas.py @tannerbeam
+sidebars.js @great-expectations/docs
```
Comment (cdkini) [inline CODEOWNERS:3]: Does this actually exist? Looks like GitHub is warning about it

File: docs_rtd/reference/spare_parts/data_context_reference.rst [hunk 2]
```diff
@@ -392,31 +392,30 @@ new directory or use this template:
           class_name: TupleFilesystemStoreBackend
           base_directory: uncommitted/data_docs/local_site/
 
-.. _Usage Statistics:
+.. _Analytics Data:
```
Comment (cdkini) [inline docs_rtd/reference/spare_parts/data_context_reference.rst:395]: This are RTD docs - could we revert these changes?

File: examples/demos/primary_keys_in_validation_results/spark/great_expectations/great_expectations.yml [hunk 3]
```diff
@@ -94,9 +94,8 @@ data_docs_sites:
     site_index_builder:
       class_name: DefaultSiteIndexBuilder
 
-anonymous_usage_statistics:
+analytics_data:
```
Comment (cdkini) [inline examples/demos/primary_keys_in_validation_results/spark/great_expectations/great_expectations.yml:97]: I think we might want to separate out the removal of the old platform and the introduction of the new one - might be too complex for a single PR?

File: great_expectations/core/usage_statistics/usage_statistics_record_schema.json [hunk 4]
```diff
@@ -1,5539 +0,0 @@
-{
-  "$schema": "https://json-schema.org/draft/2020-12/schema",
```
Comment (cdkini) [inline great_expectations/core/usage_statistics/usage_statistics_record_schema.json:2]: We can probably remove our jsonschema dependency right? Anything else to get rid off after these changes?

**Comment by jcampbell** [comment]
Handled in #9398
